var/spack/repos/builtin/packages/py-poetry/package.py

```diff
@@ -42,6 +58,14 @@ class PyPoetry(PythonPackage):
     depends_on("py-keyring@18.0.1:18", when="^python@2.7", type=("build", "run"))
     depends_on("py-keyring@20.0.1:20", when="^python@3.5", type=("build", "run"))
     depends_on("py-keyring@21.2.0:21", when="@1.1.12 ^python@3.6:3", type=("build", "run"))
-    depends_on("py-keyring@21.2.0:", when="@1.1.13 ^python@3.6:3", type=("build", "run"))
+    depends_on("py-keyring@21.2.0:", when="@1.1.13: ^python@3.6:3", type=("build", "run"))
     depends_on("py-subprocess32@3.5:3", when="^python@2.7", type=("build", "run"))
-    depends_on("py-importlib-metadata@1.6:1", when="^python@:3.7", type=("build", "run"))
+    depends_on("py-importlib-metadata@1.6:1", when="@:1.1 ^python@:3.7", type=("build", "run"))
+    depends_on("py-importlib-metadata@4.4:", when="@1.2.0 ^python@:3.9", type=("build", "run"))
+    depends_on("py-importlib-metadata@4.4:4", when="@1.2.1: ^python@:3.9", type=("build", "run"))
+    depends_on("py-jsonschema@4.10.0:4", when="@1.2:", type=("build", "run"))
+    depends_on("py-platformdirs@2.5.2:2", when="@1.2:", type=("build", "run"))
+    depends_on("py-xattr@0.9.7:0.9", when="platform=darwin @1.2:")
+    depends_on("py-urllib3@1.26.0:1", when="@1.2:")
+    depends_on("py-dulwich@0.20.44:0.20", when="@1.2.0")
+    depends_on("py-dulwich@0.20.46:0.20", when="@1.2.1:")
```

Comment:
This package would be much easier to review if the dependencies were listed in the same order as the `pyproject.toml`